extensions/true-fantom/network/network-v001.js

```diff
@@ -0,0 +1,477 @@
+            ]
+          },
+          default: { 
+            acceptReporters: true,
+            items: [
+              {
+                text: 'default',
+                value: 'default'
+              }
+            ]
+          }
+        }
+      }
+    }
+    
+    connected_to_internet_block() {
+      return navigator.onLine;
+    }
+    browser_block() {
+      let has = input => navigator.userAgent.includes(input);
+      if (has('Firefox')) return 'Firefox';
+      if (has('SamsungBrowser')) return 'Samsung Internet';
+      if ((has('Opera') || has('OPR')) && has('GX')) return 'Opera GX';
+      if (has('Opera') || has('OPR')) return 'Opera';
+      if (has('Trident')) return 'Internet Explorer';
+      if (has('Edge')) return 'Edge Legacy';
+      if (has('Edg')) return 'Edge Chromium';
+      if (has('YaBrowser') || has('YaSearchBrowser')) return 'Yandex';
+      if (has('Miui')) return 'Mi Browser';
+      if (has('UBrowser')) return 'UC Browser';
+      if (has('Chrome')) return 'Chromium';
+      if (has('Safari')) return 'Safari';
+      return '';
+    }
+    current_url_block() {
+      return document.URL || '';
+    }
+    network_type_block() {
+      let type = navigator.connection.type;
+      return type === undefined || type === 'none' || type === 'unknown' ? '' : type;
+    }
+    network_generation_block() {
+      return navigator.connection.effectiveType || '';
+    }
+    downlink_speed_block() {
+      return navigator.connection.downlink || '';
+    }
+    downlink_max_speed_block() {
+      return navigator.connection.downlinkMax || '';
+    }
+    network_rtt_block() {
+      return navigator.connection.rtt || '';
+    }
+    ping_block({SERVER}) {
+      return cached_ping_web_socket(String(SERVER));
+    }
+    get_block(args) {
+      return fetch_url(args, 'GET');
+    }
+    delete_block(args) {
+      return fetch_url(args, 'DELETE');
+    }
+    post_block(args) {
+      return fetch_url(args, 'POST');
+    }
+    put_block(args) {
+      return fetch_url(args, 'PUT');
+    }
+    patch_block(args) {
+      return fetch_url(args, 'PATCH');
+    }
+    open_link_block({URL}) {
+      window.open(URL, '_blank');
+    }
+    open_window_block({URL,WIDTH,HEIGHT,LEFT,TOP}) {
+      let params = 'popup=1';
+      params += isNaN(WIDTH) ? '' : `,width=${Number(WIDTH) < 100 ? 100 : Number(WIDTH) > window.screen.width ? window.screen.width : Number(WIDTH)}`;
+      params += isNaN(HEIGHT) ? '' : `,height=${Number(HEIGHT) < 100 ? 100 : Number(HEIGHT) > window.screen.height ? window.screen.height : Number(HEIGHT)}`;
+      params += isNaN(LEFT) ? '' : `,left=${Number(LEFT) < 0 ? 0 : Number(LEFT) > window.screen.width ? window.screen.width : Number(LEFT)}`;
+      params += isNaN(TOP) ? '' : `,top=${Number(TOP) < 0 ? 0 : Number(TOP) > window.screen.height ? window.screen.height : Number(TOP)}`;
+      window.open(URL, '_blank', params);
+    }
+    redirect_link_block({URL}) {
+      window.open(URL, '_self');
```

Comment:
This needs to protect against code execution when someone navigates to `javascript:alert(1)`

Reply:
Can you please clarify how I can do this?

Reply:
Make a URL object:

`const url = new URL(URL)`

Make sure the protocol is something that makes sense:

`if (url.protocol === 'https:' || url.protocol === 'http:' || url.protocol === 'data:' || url.protocol === 'file:') window.open(...)`